app/assets/javascripts/discourse/tests/integration/components/badge-button-test.js

```diff
@@ -77,4 +77,22 @@ module("Integration | Component | badge-button", function (hooks) {
 
     assert.dom(".user-badge.foo").exists();
   });
+
+  test("showName hides the name", async function (assert) {
```

Comment:
```suggestion
  test("setting showName to false hides the name", async function (assert) {
```

Small suggestion cause we can also set `showName` to `true`.